src/ServiceControl.MultiInstance.AcceptanceTests/SagaAudit/When_sending_saga_audit_to_main_instance.cs

```diff
@@ -66,6 +77,67 @@ public async Task Saga_history_can_be_fetched_from_main_instance()
             Assert.AreEqual("Send", sagaStateChange.InitiatingMessage.Intent);
         }
 
+        [Test]
+        public async Task
+        Check_fails_if_no_audit_retention_is_configured()
+        {
+
+            CustomServiceControlSettings = settings =>
+            {
+                settings.DisableHealthChecks = false;
+            };
+
+            CustomEndpointConfiguration = config =>
+            {
+                config.RegisterComponents(registration =>
+                {
+                    registration.ConfigureComponent((builder) =>
+                    {
+                        return new AuditRetentionCustomCheck(builder.Build<IDocumentStore>(), builder.Build<Settings>(), TimeSpan.FromSeconds(10));
+                    }, DependencyLifecycle.SingleInstance);
+                });
+            };
+
+            var scenarioContext = await WriteSagaAuditDataIntoMainInstance();
+
+            EventLogItem entry = null;
+
+            await scenarioContext
+                .Done(async c =>
+                {
+                    var result = await this.TryGetSingle<EventLogItem>("/api/eventlogitems/",
+                        e => e.EventType == "CustomCheckFailed" && e.Description.StartsWith("Saga audit data retention check"));
+                    entry = result;
+                    return result;
+                })
+                .Run();
+
+            Assert.IsTrue(entry.RelatedTo.Any(item => item == "/customcheck/Saga audit data retention check"), "Event log entry should be related to the Saga audit data retention check");
+            Assert.IsTrue(entry.RelatedTo.Any(item => item.StartsWith("/endpoint/Particular.ServiceControl")), "Event log entry should be related to the ServiceControl endpoint");
+        }
+
+        async Task<IScenarioWithEndpointBehavior<MyContext>> WriteSagaAuditDataIntoMainInstance()
+        {
+            var scenario = Define<MyContext>()
+                .WithEndpoint<SagaEndpoint>(b => b.When((bus, c) => bus.SendLocal(new MessageInitiatingSaga { Id = "Id" })));
+
+            await scenario
+                .Done(async c =>
+                {
+                    if (!c.SagaId.HasValue)
+                    {
+                        return false;
+                    }
+
+                    var sagaAudiDatatInMainInstanceIsAvailableForQuery =
+                        await this.TryGet<SagaHistory>($"/api/sagas/{c.SagaId}", instanceName: ServiceControlInstanceName);
+                    return sagaAudiDatatInMainInstanceIsAvailableForQuery;
+                })
+                .Run();
```

Comment:
Each time we run a scenario using this API the test infra will spin up a ServiceControl instances and a ServiceControl Audit instance, run the tests and then tear everything down. I'm surprised that multiple executions seem to carry the same database as I thought that the databases were all in-memory for the tests. I would consider combining this `Done` with the one above and making it a single execution. It will be slightly less readable but with some comments it should be OK.

Reply:
I think maybe because we re-use the same scenario, it all gets executed twice. i.e. We set up everything, have the endpoint send an audited saga message, store it in the database, retrieve it from the API, and then tear everything down. Then we do all of that again when we run the scenario again above.